zulipterminal/model.py

```diff
@@ -179,24 +179,40 @@ def get_message_ids_in_current_narrow(self) -> Set[int]:
             current_ids = self.index['all_starred']
         return current_ids.copy()
 
-    @asynch
-    def _start_presence_updates(self) -> None:
+    def _start_presence_initially(self) -> None:
         """
-        Notify server of user's presence each minute (version 1a)
+        Notify server of user's presence initially.
         """
-        # FIXME: version 1b: Also use 'response' to update user list
-        # FIXME: Version 2: call endpoint with ping_only=True only when
-        #        needed, and rely on presence events to update
-        while True:
-            response = self.client.update_presence(
+        self._notify_server_of_presence()
+        self._start_presence_updates()
+
+    def _notify_server_of_presence(self) -> None:
+        """
+        Notify server of user's presence every minute.
+        """
+        response = self.client.update_presence(
                 request={
                     # TODO: Determinal `status` from terminal tab focus.
                     'status': 'active' if self.new_user_input else 'idle',
                     'new_user_input': self.new_user_input,
                 }
             )
-            self.new_user_input = False
+        self.new_user_input = False
+        self.response = response
```

Comment:
Is `self.response` something you just added? Can it just be a return value?

Reply:
@neiljp I made it an attribute of the object, as it would avoid passing a huge response as a function parameter. What do you suggest here?

Reply:
The data could be large, but the returned data is just a new name for the same data, as is typical in python.

To put the question back to you: why do you think it is better to set it as an attribute?

Reply:
I was thinking of his from a C perspective (that it would be a pass-by-value here). Oops . It's a pass-by-reference.